### PR TITLE
PROJ-161 Add new RequestedActivation listener

### PR DESCRIPTION
**Pull request by LucWollants**
### Added
- Added method to get Organization based on tax number
- Added method to get Organization based on the accounting Email
- Added method to link an Organization to a Project
- Added method to create an Organization with a linked contact and organization
- Added listener for handling RequestedActivation inside the new Insightly instance

### Removed
- Removed the `getById` method for Organizations
---
Ticket: https://jira.uitdatabank.be/browse/PROJ-161

### CODE REVIEW

File: src/Integrations/Insightly/GroupIdConverter.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+declare(strict_types=1);
+
+namespace CultuurNet\ProjectAanvraag\Integrations\Insightly;
+
+use CultuurNet\ProjectAanvraag\Integrations\Insightly\ValueObjects\IntegrationType;
+use CultuurNet\ProjectAanvraag\IntegrationType\IntegrationTypeStorageInterface;
+use InvalidArgumentException;
+
+class GroupIdConverter
```
Comment (bertramakers) [inline src/Integrations/Insightly/GroupIdConverter.php:11]: Good idea! 💪

File: src/Integrations/Insightly/Serializers/ProjectSerializer.php [hunk 2]
```diff
@@ -63,15 +64,22 @@ public function toInsightlyStageChange(ProjectStage $stage): array
...
-            $this->customFieldSerializer->getCoupon($insightlyArray['CUSTOMFIELDS'])
+            $this->customFieldSerializer->getIntegrationType($insightlyArray['CUSTOMFIELDS'])
         ))->withId(
             new Id($insightlyArray['PROJECT_ID'])
         );
+
+        try {
+            $coupon = $this->customFieldSerializer->getCoupon($insightlyArray['CUSTOMFIELDS']);
+            $project = $project->withCoupon($coupon);
+        } catch (InvalidArgumentException $invalidArgumentException) {
```
Comment (bertramakers) [inline src/Integrations/Insightly/Serializers/ProjectSerializer.php:80]: When would this throw an `InvalidArgumentException` exactly? Are we not obscuring any errors here by adding a catch that does not do anything?

**Review by bertramakers — APPROVED** [review]

File: src/Integrations/Insightly/Serializers/ProjectSerializer.php (hunk 2 above)
Comment (LucWollants) [reply src/Integrations/Insightly/Serializers/ProjectSerializer.php:80]: I wanted to keep the return value of `CustomFieldSerializer::getCoupon` not nullable. So when the `Coupon` custom field is missing the `InvalidArgumentException` will be thrown. That means that the project has no `Coupon` code, but is still a valid project.

Maybe returning a nullable `Coupon` in this case would make more sense?

Comment (LucWollants) [reply src/Integrations/Insightly/Serializers/ProjectSerializer.php:80]: Very good suggestion! I use `InvalidArgumentException` way to often.

**Review by bertramakers — APPROVED** [review]

**Review by bertramakers — APPROVED** [review]

**Pull request merged**